bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/DiscoveryService.java

```diff
@@ -50,22 +51,32 @@
      * Configuration property for enabling the auto discovery feature of a
      * DiscoveryService.
      */
-    static final String CONFIG_PROPERTY_BACKGROUND_DISCOVERY = "background";
+    final String CONFIG_PROPERTY_BACKGROUND_DISCOVERY = "background";
 
     /**
      * Returns the list of {@code Thing} types which are supported by the {@link DiscoveryService}.
      *
      * @return the list of Thing types which are supported by the discovery service
      *         (not null, could be empty)
      */
-    public Collection<ThingTypeUID> getSupportedThingTypes();
+    Collection<ThingTypeUID> getSupportedThingTypes();
+
+    /**
+     * If the discovery service discovers things for a specific bridge only, this method
+     * would return the UID of the bridge.
+     *
+     * @return UID of the bridge that has started this discovery service, may be {@null}
+     */
+    default @Nullable ThingUID getBridge() {
```

Comment:
What if a discovery service _can_ discover things for several bridge instances - and you still would like to get rid of results for a single bridge?

Reply:
Isn't it a problem that a discovery service is able to discover things for more than one bridge instance? I mean, having a bridge that creates a discovery service, I only want results of this service for this bridge

Reply:
https://github.com/openhab/openhab2-addons/blob/e2befee912df9c1f65f9423a166d96a62ac8e7cc/addons/binding/org.openhab.binding.folding/src/main/java/org/openhab/binding/folding/internal/discovery/FoldingSlotDiscoveryService.java

Reply:
https://github.com/openhab/openhab2-addons/blob/739a8bb2126c22d36830b21908b051ffe3399d1e/addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/discovery/ZoneDiscoveryService.java